core/src/main/java/dev/streamx/cli/ingestion/publish/PublishCommand.java

```diff
@@ -2,86 +2,54 @@
 import dev.streamx.cli.VersionProvider;
-import dev.streamx.cli.exception.IngestionClientException;
-import dev.streamx.cli.exception.UnknownChannelException;
-import dev.streamx.cli.ingestion.IngestionArguments;
-import dev.streamx.cli.ingestion.IngestionClientConfig;
-import dev.streamx.cli.ingestion.SchemaProvider;
-import dev.streamx.cli.ingestion.StreamxClientProvider;
+import dev.streamx.cli.ingestion.BaseIngestionCommand;
 import dev.streamx.cli.ingestion.publish.payload.PayloadResolver;
 import dev.streamx.cli.ingestion.publish.payload.source.FileSourceResolver;
-import dev.streamx.cli.util.ExceptionUtils;
-import dev.streamx.clients.ingestion.StreamxClient;
 import dev.streamx.clients.ingestion.exceptions.StreamxClientException;
+import dev.streamx.clients.ingestion.publisher.Publisher;
 import jakarta.inject.Inject;
 import java.util.Collection;
 import java.util.List;
 import java.util.Optional;
 import java.util.stream.Stream;
-import javax.net.ssl.SSLHandshakeException;
 import org.jetbrains.annotations.NotNull;
 import picocli.CommandLine.ArgGroup;
 import picocli.CommandLine.Command;
-import picocli.CommandLine.Model.CommandSpec;
-import picocli.CommandLine.ParameterException;
-import picocli.CommandLine.Spec;
 
 @Command(name = PublishCommand.COMMAND_NAME,
     mixinStandardHelpOptions = true,
     versionProvider = VersionProvider.class,
     description = "Send publication data")
-public class PublishCommand implements Runnable {
+public class PublishCommand extends BaseIngestionCommand {
+
   public static final String COMMAND_NAME = "publish";
 
   @ArgGroup(exclusive = false, multiplicity = "1")
-  PublishTargetArguments ingestionTargetArguments;
-
-  @ArgGroup(exclusive = false)
-  IngestionArguments ingestionArguments;
+  PublishTargetArguments publishTargetArguments;
 
   @ArgGroup(exclusive = false)
   PayloadArguments payloadArguments;
 
-  @Spec
-  CommandSpec spec;
-
-  @Inject
-  SchemaProvider schemaProvider;
-
-  @Inject
-  StreamxClientProvider streamxClientProvider;
-
   @Inject
   PayloadResolver payloadResolver;
 
-  @Inject
-  IngestionClientConfig ingestionClientConfig;
-
   @Override
-  public void run() {
+  protected String getChannel() {
+    return publishTargetArguments.getChannel();
+  }
 
+  @Override
+  protected void perform(Publisher<JsonNode> publisher) throws StreamxClientException {
     List<PayloadArgument> mergedPayloadArgumentList = prependPayloadFile();
     JsonNode jsonNode = payloadResolver.createPayload(mergedPayloadArgumentList);
-
-    validateChannel();
-    try (StreamxClient client = streamxClientProvider.createStreamxClient()) {
-      var publisher = client.newPublisher(ingestionTargetArguments.getChannel(), JsonNode.class);
-
-      Long eventTime = publisher.publish(ingestionTargetArguments.getKey(), jsonNode);
-
-      System.out.printf("Registered data publication on '%s' at %d%n",
-          ingestionTargetArguments.getChannel(), eventTime);
```

Comment:
Lines 70-73 remain here, other code from this method is now moved to BaseIngestionCommand.

The method name is now (I think) better - action.perform instead of action.run, which was the name from Runnable class. Now the action doesn't have to _know_ it's to be run in Runnable context